components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -159,8 +204,12 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		log.Error(err, "error getting Statefulset")
 		return ctrl.Result{}, err
 	}
+
+        isImageChangeTriggerSet := metav1.HasAnnotation(instance.ObjectMeta, AnnotationImageChangeTrigger)
+        imageChangeTriggerReferencedContainerNames := getImageChangeTriggerReferencedContainerNames(instance.ObjectMeta)
```

Comment:
maybe put the two new helper functions in util.go and then refer to `imageChangeTriggerReferencedContainerNames := reconcilehelper.getImageChangeTriggerReferencedContainerNames(instance.ObjectMeta)` ? Minor thing, unsure.